extensions/slack/src/search-messages.tsx

```diff
@@ -0,0 +1,125 @@
+import { Action, ActionPanel, Color, Icon, List } from "@raycast/api";
+import { useCachedPromise, useCachedState } from "@raycast/utils";
+import { format } from "date-fns";
+import { useState } from "react";
+import * as emoji from "node-emoji";
+
+import { withSlackClient } from "./shared/withSlackClient";
+import { getSlackWebClient } from "./shared/client/WebClient";
+import { convertTimestampToDate, handleError } from "./shared/utils";
+import { useChannels, useUsers } from "./shared/client";
+import { SearchMessagesArguments } from "@slack/web-api";
+
+function Search() {
+  const [query, setQuery] = useState("");
+  const [selectedChannel, setSelectedChannel] = useState("");
+  const [sortType, setSortType] = useCachedState<SearchMessagesArguments["sort"]>("search-sort-type", "timestamp");
+
+  const { data: users } = useUsers();
+  const { data: channels } = useChannels();
+
+  const { data, isLoading } = useCachedPromise(
+    async (query, channel, sort) => {
+      const webClient = getSlackWebClient();
+      const results = await webClient.search.messages({
+        query: `${query}${selectedChannel ? ` in:${channel}` : ""}`,
+        sort,
+      });
+      return results.messages?.matches ?? [];
+    },
+    [query, selectedChannel, sortType],
+    {
+      onError(error) {
```

Comment:
thought (non-blocking): I am wondering if there is an error, should we render a `List.Empty` and make the error message more visible in addition to the toast?

Reply:
I'm not sure that's worth it here because an error could happen for subsequent searches and in this case we'd want to keep the previous search results.

Reply:
You're right, I didn't think of that edge-case. Please feel free to resolve.